jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererCodeCoverageTrackerTest.kt

```diff
@@ -318,22 +321,57 @@ class CodeWhispererCodeCoverageTrackerTest {
         verify(batcher, Times(0)).enqueue(any())
     }
 
+    @Test
```

Comment:
@andrewyuq , except the comments I left at the edit distance logic, here are some test cases you can refer to